Update README and unity doc (#1207)
ably/ably-dotnet: unity/README.md

Considerations:
* We are actively working towards automated testing by integrating Unity Cloud Build into our .NET CI pipeline.
* Installation requires developers to import a custom Unity package that includes all of Ably's dependencies.

### Supported Platforms
- Windows, MacOS, Linux, Android and iOS.

### System Requirements
* Unity 2019.x.x or newer
* The following Unity Player settings must be applied:
  * Scripting Runtime Version should be '.NET 4.x Equivalent'
  * Api Compatibility Level should be '.NET Standard 2.0'

### Downloading Unity Package
- Please download the latest Unity package from the [GitHub releases page](https://github.com/ably/ably-dotnet/releases/latest). All releases from 1.2.4 has `.unitypackage` included.

### Importing Unity Package
- Import package by going to Assets -> Import Package -> Custom Package.
  For detailed information, visit https://docs.unity3d.com/Manual/AssetPackagesImport.html
- Make sure to [disable assembly validation](CONTRIBUTING.md#disable-assembly-validation-error) if it fails due to conflict with internal newtonsoft json library.

> - Please set `ClientOptions.AutomaticNetworkStateMonitoring` to `false` in the code, since the feature is not supported and throws runtime exception.

Author: We are planning to fix it here https://github.com/ably/ably-dotnet/issues/1212

        void Start()
        {
            InitializeAbly();
        }

        private void InitializeAbly()
        {
            _clientOptions = new ClientOptions
            {
                Key = _apiKey,
                // this will disable the library trying to subscribe to network state notifications
                AutomaticNetworkStateMonitoring = false,
                AutoConnect = false,
                // this will make sure to post callbacks on UnitySynchronization Context Main Thread
                CustomContext = SynchronizationContext.Current
            };

            _ably = new AblyRealtime(_clientOptions);
            _ably.Connection.On(args =>
            {
                Debug.Log($"Connection State is <b>{args.Current}</b>");
            });
        }
    }
```
- Please take a look at [Unity Demo Chat App](./Assets/Ably/Examples/Chat/) for understanding ably SDK setup.

### Unsupported Platforms
- WebGL: Due to incompatibility with Websockets.<br/>
Read [Direct Socket Access](https://docs.unity3d.com/2019.3/Documentation/Manual/webgl-networking.html) section under WebGL Networking.

### Contributing

> - Please take a look at [contributing doc](CONTRIBUTING.md) for information related to local dev-setup, writing and running tests.

Reviewer: ```suggestion
- Please take a look at the [contributing documentation](CONTRIBUTING.md) for information relating to local development setup, and writing and running tests.
```

> - Detailed information related to updating ably DLLs using [unity plug-ins](https://docs.unity3d.com/Manual/Plugins.html) is given at [updating-ably-unitypackage](CONTRIBUTING.md#updating-ably-unitypackage).

Reviewer: ```suggestion
- Detailed information related to updating Ably DLLs using [Unity Plugins](https://docs.unity3d.com/Manual/Plugins.html) is available in [updating-ably-unitypackage](CONTRIBUTING.md#updating-ably-unitypackage).
```
